sureberus/__init__.py

```diff
@@ -32,6 +33,16 @@ def register_schemas(self, registry):
     def find_schema(self, name):
         return self.schema_registry[name]
 
+    def set_tag(self, tag, value):
+        tags = self.tags.copy()
+        tags[tag] = value
+        return attr.evolve(self, tags=tags)
+
+    def get_tag(self, tag):
```

Comment:
It seems useful to be able to supply a default if tags has not been set.

```
def get_tag(self, tag, default=TAG_DEFAULT_NOT_SET):
    if default is TAG_DEFAULT_NOT_SET:
        if tag not in self.tags:
            raise E.TagNotFound(tag, self.tags.keys(), self.stack)
        return self.tags[tag]
    else:
        return self.tags.get(tag, default)
```

Reply:
Gotcha. I've also added a `default_choice` to `when_tag_is` to match the `default_choice` in `when_key_is`, to take advantage of this.